src/components/navBar/navBar.js

```diff
@@ -0,0 +1,400 @@
+    var tabs = getTabs_();
+    for (var i = 0; i < tabs.length; i++) {
+      if (fn(tabs[i])) {
+        return tabs[i];
+      }
+    }
+
+    return null;
+  }
+
+  /**
+   * Direct focus to the selected tab when focus enters the nav bar.
+   * @export
+   */
+  function onFocus() {
+    var tab = getSelectedTab_();
+    if (tab) {
+      tab.setFocused(true);
+    }
+  }
+
+  /**
+   * Clear tab focus when focus leaves the nav bar.
+   * @export
+   */
+  function onBlur() {
+    var tab = getFocusedTab_();
+    if (tab) {
+      tab.setFocused(false);
+    }
+  }
+
+  /**
+   * Move focus from oldTab to newTab.
+   * @param {!NavLinkController} oldTab
+   * @param {!NavLinkController} newTab
+   */
+  function moveFocus(oldTab, newTab) {
+    oldTab.setFocused(false);
+    newTab.setFocused(true);
+  }
+
+  /**
+   * Responds to keypress events.
+   * @param {!jQuery.Event} e
+   * @export
+   */
+  function onKeydown(e) {
+    var tabs = getTabs_();
+    var focusedTab = googAsserts.assert(getFocusedTab_());
+    var focusedTabIndex = tabs.indexOf(focusedTab);
+
+    // use arrow keys to navigate between tabs
+    switch (e.which) {
+      case KeyCodes.UP:
+      case KeyCodes.LEFT:
+        if (focusedTabIndex > 0) {
+          moveFocus(focusedTab, tabs[focusedTabIndex - 1]);
+        }
+        break;
+      case KeyCodes.DOWN:
+      case KeyCodes.RIGHT:
+        if (focusedTabIndex < tabs.length - 1) {
+          moveFocus(focusedTab, tabs[focusedTabIndex + 1]);
+        }
+        break;
+      case KeyCodes.SPACE:
+      case KeyCodes.ENTER:
+        // timeout to avoid a "digest already in progress" console error
+        ctrl.timeout_(function() { focusedTab.getButtonEl().click(); });
+        break;
+    }
+  }
+}
+
+function MdNavLink() {
+  return {
+    restrict: 'E',
+    require: '^MdNavBar',
+    controller: MdNavLinkController,
+    bindToController: true,
+    controllerAs: 'ctrl',
+    replace: true,
+    transclude: true,
+    template:
+        '<li class="md-nav-link" role="option" aria-selected="{{ctrl.isSelected()}}">' +
+        '<md-button ng-if="ctrl.navSref" class="md-nav-button md-accent"' +
+        'ng-class="ctrl.getNgClassMap()"' +
+        'tabindex="-1"' +
+        'ui-sref="{{ctrl.navSref}}">' +
+        '<span ng-transclude></span>' +
+        '</md-button>' +
+        '<md-button ng-if="ctrl.navHref" class="md-nav-button md-accent"' +
+        'ng-class="ctrl.getNgClassMap()"' +
+        'tabindex="-1"' +
+        'ng-href="{{ctrl.navHref}}">' +
+        '<span ng-transclude></span>' +
+        '</md-button>' +
+        '<md-button ng-if="ctrl.navClick" class="md-nav-button md-accent"' +
+        'ng-class="ctrl.getNgClassMap()"' +
+        'tabindex="-1"' +
+        'ng-click="ctrl.navClick()">' +
+        '<span ng-transclude></span>' +
+        '</md-button>' +
+        '</li>',
+    scope: {
+      'navClick': '&?',
+      'navHref': '@?',
+      'navSref': '@?',
+      'linkName': '@',
```

Comment:
Yes, an external route change would have to specify which tab is now selected. Something like this:
```
$rootScope.$on('$routeChangeSuccess', function(event, current) {
    $scope.currentLink = getCurrentLinkFromRoute(current);
});
```
Where the user is responsible for the details of `getCurrentLinkFromRoute`, but it could include logic like "return 'admin' if current route is a child of /admin".

I think the main issue is that somehow the user has to tell us what tab should be selected. If we remove link-name, we need to replace it with another method for saying "select this tab".

I suppose we could use index, but that does not seem better - individual tabs could be ng-if'd (I have seen some examples of this) and so the tab indices could change over time.